packages/web/src/pages/api/rewards/new.js

```diff
@@ -0,0 +1,59 @@
+// Local imports
+import {
+	auth,
+	firebase,
+	firestore,
+} from 'helpers/firebase.admin'
+import { createEndpoint } from 'helpers/createEndpoint'
+import httpStatus from 'helpers/httpStatus'
+
+
+
+
+
+export const handler = async (request, response) => {
+	const reward = { ...request.body }
+	const { firebaseAuthToken } = request.cookies
+
+	try {
+		const user = await auth.verifyIdToken(firebaseAuthToken, true)
+
+		if (!reward.title || !reward.cost || reward.uid !== user.uid) {
+			response.status(httpStatus.UNPROCESSABLE_ENTITY).end()
+		}
+        
+		const rewardsCollection = firestore
```

Comment:
I think @Marenthyu and I discussed this on Discord, but it never made it into the ticket. Instead of creating the rewards in a subcollection of the `Campaign`, we should create them in a new top-level `rewards` collection and assign them a `campaignID` field. This will make it much easier (and more cost effective) for the sync API to find all of the info it needs.

Reply:
It was discussed, aye - reasoning being that to listen to ANY changes, I'd need to listen to changes for ALL subcollections, which is bull**** for scalability. With a top-level colection, a single listener does the trick.

Reply:
This should be fixed now.